port/cpl_string.cpp

```diff
@@ -3161,3 +3161,56 @@ int CPLTolower(int c)
 {
     return (c >= 'A' && c <= 'Z') ? (c - 'A' + 'a') : c;
 }
+
+/************************************************************************/
+/*                      CPLRemoveSQLComments()                          */
+/************************************************************************/
+
+/** Remove SQL comments from a string
+ *
+ * @param osInput Input string.
+ * @since GDAL 3.11
+ */
+std::string CPLRemoveSQLComments(const std::string &osInput)
+{
+    const CPLStringList aosLines(
+        CSLTokenizeStringComplex(osInput.c_str(), "\r\n", FALSE, FALSE));
+    std::string osSQL;
+    for (const char *pszLine : aosLines)
+    {
+        char chQuote = 0;
+        int i = 0;
+        for (; pszLine[i] != '\0'; ++i)
+        {
+            if (chQuote)
+            {
+                if (pszLine[i] == chQuote)
+                {
+                    if (pszLine[i + 1] == chQuote)
```

Comment:
What is this case? You already have an opening quote. Now you're checking for two in a row? Comment?

Could the advance of `i` put you past the end of the string because you're also advancing in the for loop?

Reply:
comment ``// Deal with escaped quote character which is repeated, like in 'foo''bar' or "foo""bar".`` added

> Could the advance of i put you past the end of the string because you're also advancing in the for loop?

No that should be fine. In that case, it is expected to advance i of 2 because we have processed 2 characters at once

Reply:
yes, probably. I'd say it is a matter of taste. At least that's the way similar code that deals with quoted strings is written elsewhere in the code base, especially when you need to emit the unescaped version without the leading/trailing quotes, so I feel it is better to keep the same style.